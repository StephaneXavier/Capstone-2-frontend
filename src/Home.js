import React, { useState,  } from "react";
import { Button } from "reactstrap";
import WashroomApi from "./api";
import { findClosestCityWashroom, userDeniesLocation, compareCityAndDBWashrooms } from './helpers'

const GOOGLE_URL = 'https://www.google.com/maps/search/'

const Home = ({ cityWashrooms }) => {

    const [nearestWashroom, setNearestWashroom] = useState();
    const [userLocation, setUserLocation] = useState();


    function userSharesLocation(position) {
        const { longitude, latitude } = position.coords
        async function findNearestWashroom() {
            const nearestCityWashroom = findClosestCityWashroom(longitude, latitude, cityWashrooms);
            const nearestDBWashroom = await WashroomApi.getClosestWashroom(longitude, latitude);

            const nearestWashroomObj = compareCityAndDBWashrooms({ longitude, latitude }, nearestDBWashroom, nearestCityWashroom);
            const nearestWashroomCoords = nearestWashroomObj.geometry ?
                { latitude: nearestWashroomObj.geometry.y, longitude: nearestWashroomObj.geometry.x }
                :
                { latitude: nearestWashroomObj.latitude, longitude: nearestWashroomObj.longitude };

            setNearestWashroom(nearestWashroomCoords)

        }
        findNearestWashroom()
    }


    function onClick() {
        const options = { timeout: 10000 };
        window.navigator.geolocation.getCurrentPosition(userSharesLocation, userDeniesLocation, options)
    }

    function resetLocation(){
        setNearestWashroom(null)
    }

    return (
        <div>

            {cityWashrooms ?
                <>

                    {
                        nearestWashroom ?
                            <>
                                <a href={`${GOOGLE_URL + nearestWashroom.latitude + ' ' + nearestWashroom.longitude}`} > Nearest Washroom </a> <br></br>
                                <Button onClick={resetLocation}> reset location </Button>
                            </>


                            :
                            <Button className='rounded' onClick={onClick}>Find a washroom!</Button>
                    }

                </>
                :

                <h1>fetching data...</h1>

            }

        </div>
    )
}

export default Home