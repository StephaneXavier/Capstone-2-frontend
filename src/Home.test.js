import React from "react";
import Home from "./Home";
import { fireEvent, render } from "@testing-library/react";
import '@testing-library/jest-dom'

test('it renders without crashing', () => {
    render(<Home/>)
})


test('it renders without crashing', () => {
    render(<Home/>)
})

it('should keep the same snapshot', () => {
    const {asFragment} = render(<Home/>);
    expect(asFragment()).toMatchSnapshot();
});

it('should show fetching data when page first loads',() => {
    const {queryByText, debug} = render(<Home/>)
    expect(queryByText("fetching data...")).toBeInTheDocument()
})

it('should show the find a washroom button once API data is done loading', () => {
    
})