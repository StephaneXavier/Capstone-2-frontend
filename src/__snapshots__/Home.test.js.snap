// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should keep the same snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      fetching data...
    </h1>
  </div>
</DocumentFragment>
`;
